storagemarket/impl/client_storagemarket.go

```diff
@@ -87,21 +86,21 @@ func (c *Client) ProposeStorageDeal(
 	addr address.Address,
 	info *storagemarket.StorageProviderInfo,
 	data *storagemarket.DataRef,
-	proposalExpiration storagemarket.Epoch,
-	duration storagemarket.Epoch,
-	price tokenamount.TokenAmount,
-	collateral tokenamount.TokenAmount,
+	startEpoch abi.ChainEpoch,
+	endEpoch abi.ChainEpoch,
+	price abi.TokenAmount,
+	collateral abi.TokenAmount,
 ) (*storagemarket.ProposeStorageDealResult, error) {
 
 	proposal := ClientDealProposal{
-		Data:               data,
-		PricePerEpoch:      price,
-		ProposalExpiration: uint64(proposalExpiration),
-		Duration:           uint64(duration),
-		Client:             addr,
-		ProviderAddress:    info.Address,
-		MinerWorker:        info.Worker,
-		MinerID:            info.PeerID,
+		Data:            data,
+		PricePerEpoch:   price,
+		StartEpoch:      startEpoch,
```

Comment:
We'll need to make sure we guard against negative epochs now too.

Reply:
I feel like the check to see if StartEpoch > current chain height protects against that, but I wish we were using unsigned ints in spec